qubesbuilder/plugins/__init__.py

```diff
@@ -0,0 +1,377 @@
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2021 Frédéric Pierret (fepitre) <frederic@invisiblethingslab.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from pathlib import Path, PurePosixPath
+from typing import List, Dict
+
+import dateutil.parser
+import yaml
+from dateutil.parser import parse as parsedate
+
+from qubesbuilder.component import QubesComponent
+from qubesbuilder.distribution import QubesDistribution
+from qubesbuilder.template import QubesTemplate
+
+# Directories inside executor
+BUILDER_DIR = Path("/builder")
+BUILD_DIR = BUILDER_DIR / "build"
+PLUGINS_DIR = BUILDER_DIR / "plugins"
+DISTFILES_DIR = BUILDER_DIR / "distfiles"
+REPOSITORY_DIR = BUILDER_DIR / "repository"
+CACHE_DIR = BUILDER_DIR / "cache"
+
+
+class PackagePath(PurePosixPath):
+    def mangle(self):
+        return str(self).replace("/", "_")
```

Comment:
This isn’t reversible, so it is possible that different paths could have the same mangled name.  I recommend ensuring that the mangling is reversible and can never produce `.` or `..`.